### PR TITLE
Remove non-bootstrap packages from bootstrap tarball

### PR DESCRIPTION
**Pull request by branden**
## High Level Description

This PR removes all packages from the bootstrap tarball except the minimum necessary to install the Pkgpanda CLI. It also adds all cluster packages to `/etc/mesosphere/setup-flags/cluster-packages.json`, in addition to the config packages that were already listed there. This directs Pkgpanda to fetch any packages not present in the bootstrap tarball during node setup. In order to make packages available for fetch in an onprem deployment, the installer now outputs them as individual artifacts under `genconf/serve/`.

## Related Issues

 - [DCOS-7832](https://jira.mesosphere.com/browse/DCOS-7832) Explode bootstrap tarball

## Checklist for all PR's

  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]

### CODE REVIEW

**Comment by adammesos** [comment]
Is this ready for review? If/when it is, please assign a couple of reviewers.

**Comment by branden** [comment]
@amitaekbote @BenWhitehead @spahl This PR isn't ready for a final review yet, but it's mostly finished. I'm still working on deduping the `cluster_packages` list in CF templates to get them under the size limit. Everything else should be pretty settled.

**Comment by branden** [comment]
@amitaekbote ?

**Comment by amitaekbote** [comment]
ah weird, the comment doesnt point to the commit with all the tests moved, will we be moving the tests or will you remove that commit?

**Comment by branden** [comment]
Not sure, yet to debug.

File: dcos_installer/config_util.py [hunk 1]
```diff
@@ -41,6 +41,11 @@ def rollback():
     try:
         # Copy across
         for filename in filenames:
+            # TODO(branden): Need to clean this up on rollback
```
Comment (amitaekbote) [inline dcos_installer/config_util.py]: Can we fix this with this pr?

Comment (branden) [inline dcos_installer/config_util.py]: Planning on it.

**Comment by branden** [comment]
@amitaekbote @BenWhitehead @spahl Ready for review.

File: release/__init__.py [hunk 2]
```diff
@@ -272,23 +272,36 @@ def get_gen_package_artifact(package_id_str):
         'local_path': package_filename}
 
 
-def make_bootstrap_artifacts(bootstrap_id, variant_name, artifact_prefix):
+def make_bootstrap_artifacts(bootstrap_id, package_ids, variant_name, artifact_prefix):
     bootstrap_filename = "{}.bootstrap.tar.xz".format(bootstrap_id)
+    bootstrap_filename = "{}.bootstrap.tar.xz".format(bootstrap_id)
```
Comment (BenWhitehead) [inline release/__init__.py]: duplicate line

**Comment by branden** [comment]
@mesosphere-mergebot bump-ee

**Comment by branden** [comment]
@mesosphere-mergebot bump-ee

**Comment by branden** [comment]
@amitaekbote @BenWhitehead Please re-review commit eeb3a970dfa266dd77516c86f2c4f7fde465e4a7. I updated the `package_ids` param to pull in the correct `complete.latest.json` artifact for downstream variants.

**Pull request merged**